homeassistant/components/viaris/sensor.py

```diff
@@ -0,0 +1,1207 @@
+
+
+class ViarisSensorMennekes2(ViarisEntity, SensorEntity):
+    """Representation of the Viaris portal."""
+
+    entity_description: ViarisSensorEntityDescription
+
+    def __init__(
+        self,
+        config_entry: config_entries.ConfigEntry,
+        description: ViarisSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(config_entry, description)
+
+        self.entity_description = description
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._attr_native_value is not None
+
+    async def async_added_to_hass(self) -> None:
+        """Publish mennekes and subscribe MQTT events."""
+
+        @callback
+        def message_received_mennekes2(message):
+            """Handle new MQTT messages."""
+            if self.entity_description.state is not None:
+                self._attr_native_value = self.entity_description.state(message.payload)
+
+            else:
+                self._attr_native_value = message.payload
+
+            self.async_write_ha_state()
+
+        if self._model == MODEL_COMBIPLUS:
+            await mqtt.async_subscribe(
+                self.hass,
+                self._topic_evsm_mennekes2_subs,
+                message_received_mennekes2,
+                0,
+            )
+            await mqtt.async_subscribe(
+                self.hass,
+                self._topic_evsm_menek2_value_subs,
+                message_received_mennekes2,
+                0,
+            )
+            value = {"idTrans": 0}
+            value_json = json_dumps(value)
+            await mqtt.async_publish(
+                self.hass, self._topic_evsm_mennekes2_pub, value_json
+            )
+
+
+class ViarisSensorMqttCfg(ViarisEntity, SensorEntity):
+    """Representation of the Viaris portal."""
+
+    entity_description: ViarisSensorEntityDescription
+
+    def __init__(
+        self,
+        config_entry: config_entries.ConfigEntry,
+        description: ViarisSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(config_entry, description)
+
+        self.entity_description = description
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._attr_native_value is not None
+
+    async def async_added_to_hass(self) -> None:
+        """Publish mqtt config and subscribe MQTT events."""
+
+        @callback
+        def message_received_mqtt_cfg(message):
+            """Handle new MQTT messages."""
+            if self.entity_description.state is not None:
+                self._attr_native_value = self.entity_description.state(message.payload)
+
+            else:
+                self._attr_native_value = message.payload
+
+            self.async_write_ha_state()
+
+        await mqtt.async_subscribe(
+            self.hass, self._topic_mqtt_subs, message_received_mqtt_cfg, 0
```

Comment:
Look at how tasmota or mysensors integration use the mqtt integration. The 3rd party library should handle all the MQTT details.